### PR TITLE
 [fix] weather manager not ready if settings get loaded for the first time

### PR DESCRIPTION
**Pull request by xhaggi**
see title

Issue reported at https://forum.kodi.tv/showthread.php?tid=298461&pid=2687412#pid2687412

@MilhouseVH

### CODE REVIEW

File: xbmc/ServiceBroker.cpp [hunk 1]
```diff
@@ -142,6 +142,11 @@ bool CServiceBroker::IsBinaryAddonCacheUp()
   return g_application.m_ServiceManager->init_level > 1;
 }
 
+bool CServiceBroker::IsServiceManagerUp()
```
Comment (ksooo) [inline xbmc/ServiceBroker.cpp]: This method could be `const`.

Comment (ksooo) [reply xbmc/ServiceBroker.cpp]: Ooops, overlooked its `static`. Sorry.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by MilhouseVH** [comment]
Thanks @xhaggi - tested by changing the majority of regional settings and no more crashes, so looks good.

**Comment by xhaggi** [comment]
@notspiff mind taking a look

**Review by notspiff — APPROVED** [review]

**Comment by xhaggi** [comment]
Thx

**Comment by scott967** [comment]
This should fix the crash I entered trac ticket #17706  Didn't see the post on Milhouse's forum.

**Pull request merged**